### DISCUSSION TITLE
📕Documentation Improvements - I need your help🙏

### DISCUSSION THREAD

**Discussion by ahochsteger** [ahochsteger/gmail-processor#352] [General]
Recently I've [put some effort](https://github.com/ahochsteger/gmail-processor/pull/336) into behind-the-scenes automation to improve the maintenance of examples that enables:
* automatic generation of the [example documentation pages](https://ahochsteger.github.io/gmail-processor/docs/examples/)
* automatic generation of Jest spec files for unit testing (see `*.spec.ts` files in sub-folders of [this source examples folder](https://github.com/ahochsteger/gmail-processor/tree/main/src/examples))
* automatic generation of end-to-end tests that can be run directly within Google Apps Script (see `*.js` files in sub-folders of [this GAS examples folder](https://github.com/ahochsteger/gmail-processor/tree/main/src/gas/examples)).
* selection of examples in the [playground](https://ahochsteger.github.io/gmail-processor/playground)

Many other parts of the documentation are already automatically generated as much as possible to make sure it is always up-to-date, like:
* [Config Schema](https://ahochsteger.github.io/gmail-processor/docs/reference/config-schema)
* [Enum Types](https://ahochsteger.github.io/gmail-processor/docs/reference/enum-types) that are used to define possible values for configuration fields
* [Actions](https://ahochsteger.github.io/gmail-processor/docs/reference/actions)
* [Substitution Placeholder](https://ahochsteger.github.io/gmail-processor/docs/reference/placeholder)
* [Logging Configuration](https://ahochsteger.github.io/gmail-processor/docs/reference/logging)

All this is good, but I feel it still can get better - especially for new users.
That's where I need your help 🙏.
As a user of Gmail Processor there are many ways you can help to improve the documentation like:
* Suggest topics, content or examples you'd expect but did not find
* Describe your use cases to inspire others where Gmail Processor can be of any help
* Contribute [documentation](https://github.com/ahochsteger/gmail-processor/tree/main/docs/docs) or [examples](https://github.com/ahochsteger/gmail-processor/tree/main/src/examples) by [forking gmail-processor](https://github.com/ahochsteger/gmail-processor/fork)
* Anything else that may help users

Thanks in advance for any help!
Andreas

**Comment by tepedersen** [comment]
The Doc under Exampes/Advanced/Regular Expression might be even better if amended with a quick sentence on how to enable case-insensitive matches. 

Spent the longest time trying to get "/.*searchword.*/i" to match, only after I looked through the code for Regexutils did I find out that flags like 'i' should be given as "(?i)searchword" instead, which seems to be a library specific implementation. Now it works great though! 

Keep up the great work!

**Reply by ahochsteger** [reply]
@tepedersen thanks for your great feedback - there's now a note at the [RegEx example page](https://ahochsteger.github.io/gmail-processor/docs/examples/advanced/regularExpressions).
